front/pages/api/w/[wId]/data_sources/[dsId]/files.ts

```diff
@@ -124,4 +120,57 @@ async function handler(
   }
 }
 
+function handlePublicErrorResponse(req: any, res: any, error: DustError) {
+  let status_code: number;
+  let type: APIErrorType;
+  let panic: boolean;
+
+  switch (error.code) {
+    case "file_not_ready":
+    case "invalid_file":
+    case "title_too_long":
+    case "invalid_url":
+    case "missing_csv":
+    case "invalid_content_error":
+      status_code = 400;
+      type = "invalid_request_error";
+      panic = false;
+      break;
+
+    case "unauthorized":
+      status_code = 401;
+      type = "not_authenticated";
+      panic = false;
+      break;
+
+    case "resource_not_found":
+      status_code = 404;
+      type = "file_not_found";
+      panic = false;
+      break;
+
+    case "data_source_quota_error":
+      status_code = 413;
+      type = "file_too_large";
```

Comment:
This is not a file_too_large but rather `data_source_quota_error` which exists in API error as well